docker/travis_test/prepare/00-system_deps.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash -xv
+
+echo "#######################################################"
+echo "#          Starting build tools installation          #"
+echo "#######################################################"
+
+zypper ar -p 1 -f -G "https://download.opensuse.org/repositories/devel:/openQA:/Leap:/42.3/openSUSE_Leap_42.3/devel:openQA:Leap:42.3.repo"
+zypper ref
+zypper dup -y
+
+zypper in -y -C automake \
+       curl \
+       dbus-1-devel \
+       fftw3-devel \
+       gcc \
+       gcc-c++ \
+       git \
+       gmp-devel \
+       gzip \
+       libexpat-devel \
+       libsndfile-devel \
+       libssh2-1 \
+       libssh2-devel \
+       libtheora-devel \
+       libtool \
+       libxml2-devel \
+       make \
+       opencv-devel \
+       patch \
+       postgresql-devel \
+       qemu \
+       qemu-tools \
+       qemu-kvm \
+       tar \
+       optipng \
+       sqlite3 \
+       postgresql-server \
+       which \
+       chromedriver \
+       'rubygem(sass)' \
+        perl \
+       emacs
```

Comment:
geez, do we really need to cover all these manually in a shell script?

Reply:
well, emacs isn't required but i need an editor to edit some files. I'll remove the emacs dependency and i'll create a new image.

Reply:
yeah, but I mean *all of it*. So we have dependencies mentioned already in:

* docs
* spec files of the openSUSE project
* cpanfile

* and now in shell scripts?

I don't know the perfect solution I am just worried that this looks hard to maintain

Reply:
Isn't there a way to extrapolate revdeps out of zypper commands?

Reply:
We need a way to create the image. 

> docs

The dependecies are not mentioned. It just says to install openqa with `zypper in openQA`

> spec files of the openSUSE project

Well i guess parsing the spec files would be possible.

> cpanfile

These are just the perl packages. Not the system deps. But we already use the cpanfile to test if all the packages are present. So to use the cpanfile we would need to change the requirements.

If some dependency is added to the code and not to the image, the tests will fail.

Reply:
@okurz Let's go baby steps here. If a new dependency is added, tests will fail and we'll notice for sure (Could be painful). But in any case, I see many follow ups to this work. But we need a stepping stone :)

Reply:
@mudler 

maybe with `zypper info --requires openqa`

> Information for package openQA:
> -------------------------------
> Repository     : openQA                                      
> Name           : openQA                                      
> Version        : 4.5.1528115270.6c0f9c0e-111.1               
> Arch           : noarch                                      
> Vendor         : obs://build.opensuse.org/devel:openQA       
> Installed Size : 6.9 MiB                                     
> Installed      : No                                          
> Status         : not installed                               
> Source package : openQA-4.5.1528115270.6c0f9c0e-111.1.src    
> Summary        : The openQA web-frontend, scheduler and tools
> Description    :                                             
>     openQA is a testing framework that allows you to test GUI applications on one
>     hand and bootloader and kernel on the other. In both cases, it is difficult to
>     script tests and verify the output. Output can be a popup window or it can be
>     an error in early boot even before init is executed.
> 
>     openQA is an automated test tool that makes it possible to test the whole
>     installation process of an operating system. It uses virtual machines to
>     reproduce the process, check the output (both serial console and screen) in
>     every step and send the necessary keystrokes and commands to proceed to the
>     next. openQA can check whether the system can be installed, whether it works
>     properly in 'live' mode, whether applications work or whether the system
>     responds as expected to different installation options and commands.
> 
>     Even more importantly, openQA can run several combinations of tests for every
>     revision of the operating system, reporting the errors detected for each
>     combination of hardware configuration, installation options and variant of the
>     operating system.
> Requires       : [17]                                        
>     /usr/bin/perl
>     /bin/bash
>     git-core
>     optipng
>     dbus-1
>     perl(LWP::Protocol::https)
>     perl(URI)
>     openQA-client = 4.5.1528115270.6c0f9c0e
>     openQA-common = 4.5.1528115270.6c0f9c0e
>     perl(DBIx::Class) >= 0.082801
>     perl(Minion) >= 9.02
>     perl-Mojolicious-Plugin-AssetPack = 2.03
>     /bin/sh
>     coreutils
>     systemd
>     perl(SQL::SplitStatement)
>     perl(DBIx::Class::DeploymentHandler)


?

Reply:
the package's dependencies are maintained in OBS - which is something we should fix just as well. the spec file should be in git - and possibly generated from a common format.